snuba/subscriptions/scheduler_consumer.py

```diff
@@ -243,6 +249,70 @@ def __build_tick_consumer(self) -> CommitLogTickConsumer:
         )
 
 
+class MeasureCommitLogOrderMetrics(ProcessingStrategy[KafkaPayload]):
+    """
+    This is an extremely minimalistic strategy that does absolutely nothing
+    except measure the extent to which commits are monotonically ordered
+    by partition in the snuba-commit-log.
+
+    It is intended to be temporary code to help debug some invalid interval
+    issues with the tick consumer in production.
+    """
+
+    def __init__(
+        self,
+        partitions: int,
+        metrics: MetricsBackend,
+        commit: Callable[[Mapping[Partition, Position]], None],
+    ) -> None:
+        self.__previous_messages: MutableMapping[Partition, MessageDetails] = {}
+        self.__partitions = partitions
+        self.__metrics = metrics
+        self.__commit = commit
+
+    def poll(self) -> None:
+        pass
+
+    def submit(self, message: Message[KafkaPayload]) -> None:
+        commit = commit_codec.decode(message.payload)
+
+        assert commit.partition.index < self.__partitions
+        assert commit.orig_message_ts is not None
+
+        current_message = MessageDetails(commit.offset, commit.orig_message_ts)
+
+        # Record metrics about in order vs out of order messages
+        last_message = self.__previous_messages.get(commit.partition)
+        if last_message is not None:
+            current_message = MessageDetails(commit.offset, commit.orig_message_ts)
+            try:
+                Interval(last_message.orig_message_ts, current_message.orig_message_ts)
+                self.__metrics.increment(
+                    "build_interval",
+                    tags={"valid": "true", "partition": str(commit.partition.index)},
+                )
+            except InvalidRangeError:
+                self.__metrics.increment(
+                    "build_interval",
+                    tags={"valid": "false", "partition": str(commit.partition.index)},
+                )
+
+        # Update self.__previous_messages
+        self.__previous_messages[commit.partition] = current_message
+
+        # Commit the offset for every message no matter what
+        self.__commit({message.partition: Position(message.offset, message.timestamp)})
+
+    def close(self) -> None:
+        pass
+
+    def terminate(self) -> None:
+        pass
+
+    def join(self, timeout: Optional[float] = None) -> None:
+        pass
```

Comment:
You may want to call the next step methods even if this is temporary code, just to be consistent.